web/app.js

```diff
@@ -2098,9 +2098,13 @@ const PDFViewerApplication = {
   unbindWindowEvents() {
     this._windowAbortController?.abort();
     this._windowAbortController = null;
-    if (AppOptions.get("isInAutomation")) {
+    if (
+      (typeof PDFJSDev !== "undefined" && PDFJSDev.test("TESTING")) ||
+      AppOptions.get("isInAutomation")
+    ) {
       this._globalAbortController?.abort();
       this._globalAbortController = null;
+      this.pdfHistory?.reset();
```

Comment:
Can you please call and await `PDFViewerApplication.close` in the tests, rather than adding this line "randomly" here?